controller/src/test/java/io/pravega/controller/server/eventProcessor/RequestHandlersTest.java

```diff
@@ -771,6 +777,28 @@ public void testDeleteAssociatedStream() {
         assertFalse(streamStore.checkStreamExists(scope, markStream, null, executor).join());
     }
 
+    @Test
+    public void testDeleteScopeRecursive() throws Exception {
+        final String scopeName = "deleteScope";
+        final String testScopeName = "testScope";
+        // Create a scope
+        streamStore.createScope(scopeName, null, executor).join();
+        // Verify that the scope is created
+        assertTrue(streamStore.checkScopeExists(scope, null, executor).join());
+        assertFalse(streamStore.checkScopeExists(testScopeName, null, executor).join());
+
+        // Instantiate DeleteScopeEvent and DeleteScopeTask
+        DeleteScopeTask deleteScopeTask = new DeleteScopeTask(streamMetadataTasks, streamStore, kvtStore, executor);
+        DeleteScopeEvent deleteScopeEvent = new DeleteScopeEvent(scopeName, 0L);
+
+        // Submit the execute method of DeleteScopeTask
+        deleteScopeTask.execute(deleteScopeEvent).join();
+        Thread.sleep(1000);
```

Comment:
Lets not add sleeps to tests. This should not be needed.

Reply:
sleep is removed from the test now